lib/NativeCode/DynamicLibraryLoaderHelper/NativeRender/src/eos_helpers.cpp

```diff
@@ -243,170 +300,166 @@ namespace pew::eos
-            steam_integrated_platform_option.Type = EOS_IPT_Steam;
-            steam_integrated_platform_option.Flags = eos_steam_config.flags;
-            steam_integrated_platform_option.InitOptions = &steam_platform;
+        platform_options.ProductId = product_config.product_id.c_str();
+        platform_options.SandboxId = platform_config.deployment.sandbox.id.c_str();
+        platform_options.DeploymentId = platform_config.deployment.id.c_str();
+        platform_options.ClientCredentials.ClientId = platform_config.client_credentials.client_id.c_str();
+        platform_options.ClientCredentials.ClientSecret = platform_config.client_credentials.client_secret.c_str();
 
-            steam_platform.ApiVersion = EOS_INTEGRATEDPLATFORM_STEAM_OPTIONS_API_LATEST;
+        platform_options.TickBudgetInMilliseconds = platform_config.tick_budget_in_milliseconds;
 
-            EOS_IntegratedPlatform_CreateIntegratedPlatformOptionsContainerOptions options = { EOS_INTEGRATEDPLATFORM_CREATEINTEGRATEDPLATFORMOPTIONSCONTAINER_API_LATEST };
-            eos_library_helpers::EOS_IntegratedPlatform_CreateIntegratedPlatformOptionsContainer_ptr(&options, &integrated_platform_options_container);
-            platform_options.IntegratedPlatformOptionsContainerHandle = integrated_platform_options_container;
-
-            EOS_IntegratedPlatformOptionsContainer_AddOptions addOptions = { EOS_INTEGRATEDPLATFORMOPTIONSCONTAINER_ADD_API_LATEST };
-            addOptions.Options = &steam_integrated_platform_option;
-            eos_library_helpers::EOS_IntegratedPlatformOptionsContainer_Add_ptr(integrated_platform_options_container, &addOptions);
+        if (platform_config.task_network_timeout_seconds > 0)
+        {
+            double task_network_timeout_seconds_dbl = platform_config.task_network_timeout_seconds;
+            platform_options.TaskNetworkTimeoutSeconds = &task_network_timeout_seconds_dbl;
         }
-#endif
 
-        //EOS_Platform_Options_debug_log(platform_options);
-        logging::log_inform("run EOS_Platform_Create");
+        apply_rtc_options(platform_options);
+        apply_steam_settings(platform_options);
+
+        return platform_options;
+    }
+
+    PEW_EOS_API_FUNC(EOS_Platform_Options*) PEW_EOS_Get_CreateOptions()
+    {
+        const auto platform_config = Config::get<WindowsConfig>();
+        const auto product_config = Config::get<ProductConfig>();
+
+        const auto create_options = get_create_options(*platform_config, *product_config);
+
+        // Allocate a new pointer so that it can be returned instead of the
+        // pointer to the object created on the stack above.
+        const auto platform_options_ptr = new EOS_Platform_Options();
+        *platform_options_ptr = create_options;
+
+        return platform_options_ptr;
+    }
+
+    PEW_EOS_API_FUNC(EOS_InitializeOptions*) PEW_EOS_Get_InitializeOptions()
```

Comment:
You are correct, this was a problem I discovered and subsequently fixed in another PR where I make use of these functions to test equivalency with managed implementation. Since these functions should only be used in testing, I corrected the implementation _and_ enclosed both their definitions and implementations in a `_DEBUG` flag to make sure they don't get included in any released binaries.